src/main/java/org/spongepowered/common/item/inventory/custom/CustomInventory.java

```diff
@@ -138,36 +141,13 @@ public Container createContainer(InventoryPlayer playerInventory, EntityPlayer p
         // TODO custom container including filters and other slot stuff
         this.viewers.add(playerIn);
 
-        return new CustomContainer(playerIn, this);
+        return ((Container) this.archetype.getContainer(((Inventory) this), ((Player) playerIn)));
     }
 
     @Override
     public String getGuiID() {
-        if (this.archetype == InventoryArchetypes.CHEST || this.archetype == InventoryArchetypes.DOUBLE_CHEST) {
-            return "minecraft:chest";
-        } else if (this.archetype == InventoryArchetypes.HOPPER) {
-            return "minecraft:hopper";
-        } else if (this.archetype == InventoryArchetypes.DISPENSER) {
-            return "minecraft:dispenser";
-        } else if (this.archetype == InventoryArchetypes.WORKBENCH) {
-            return "minecraft:crafting_table";
-        } else if (this.archetype == InventoryArchetypes.FURNACE) {
-            return "minecraft:furnace";
-        } else if (this.archetype == InventoryArchetypes.ENCHANTING_TABLE) {
-            return "minecraft:enchanting_table";
-        } else if (this.archetype == InventoryArchetypes.ANVIL) {
-            return "minecraft:anvil";
-        } else if (this.archetype == InventoryArchetypes.BREWING_STAND) {
-            return "minecraft:brewing_stand";
-        } else if (this.archetype == InventoryArchetypes.BEACON) {
-            return "minecraft:beacon";
-        } else if (this.archetype == InventoryArchetypes.HORSE || this.archetype == InventoryArchetypes.HORSE_WITH_CHEST) {
-            return "EntityHorse";
-        } else if (this.archetype == InventoryArchetypes.VILLAGER) {
-            return "minecraft:villager";
-        } else {
-            return "minecraft:chest";
-        }
+        return this.archetype.getProperty(GuiIDProperty.class, GuiIDProperty.class.getSimpleName().toLowerCase()).map(p -> p.getValue()).orElse
+                ("minecraft:chest");
```

Comment:
This formatting is weird - you should move the entire call (`.orElse("minecraft:chest");` to this line, or move this to the previous line.